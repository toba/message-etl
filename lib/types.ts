export enum Source {
   FacebookMessenger,
   GoogleVoice,
   GMail
}

export enum Relation {
   None = 0,
   Self,
   Other
}

export interface Message {
   /** Who sent the message. */
   from: Relation;
   /** When the message was sent. */
   on: Date;
   subject?: string;
   /** The platform used to convey the message. */
   source: Source;
   /** Content of the message. */
   text: string | null;
   /** An emoji reaction. */
   reaction?: string;
}

/**
 * Methods to process and parse messages from a particular platform.
 */
export interface Reader {
   sourceName: string;
   /** Filter file list so it only includes those for the platform. */
   filter: (fileName: string) => boolean;
   /** Process an exported file. */
   process: (fileText: string) => Message[];
   /** Parse a single message from the exported file. */
   parse: (msg: any) => Message | null;
}

export interface Writer {
   serialize: (m: Message) => string;
   save: (messages: Message[]) => boolean;
}

export interface Person {
   name: string;
   alias: string[];
   email: string[];
   phone: string[];
}

export type People = { [key: number]: Person };

export namespace Facebook {
   export interface Reaction {
      /** Emoji */
      reaction: string;
      /** Person's name */
      actor: string;
   }

   export interface Link {
      uri: string;
   }

   export interface MediaLink extends Link {
      creation_timestamp: number;
   }

   export interface VideoLink extends MediaLink {
      thumbnail: Link;
   }

   export interface Message {
      sender_name: string;
      timestamp_ms: number;
      content: string;
      type: 'Generic';
      sticker?: Link;
      videos?: VideoLink[];
      photos?: MediaLink[];
      reactions?: Reaction[];
   }

   interface Participant {
      name: string;
   }

   /**
    * Exported Facebook Messenger data. Export files are per-conversation.
    * @see https://www.facebook.com/help/1701730696756992?helpref=hc_global_nav
    */
   export interface Conversation {
      participants: Participant[];
      messages: Message[];
      title: string;
      is_still_participant: boolean;
      thread_type: string;
      thread_path: string;
   }
}
