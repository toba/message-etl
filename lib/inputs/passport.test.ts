import '@toba/test';
import { readFileText } from '@toba/test';
import { passport } from './index';
import { Relation } from '../types';

jest.mock('../config/index');

test('matches file names', () => {
   ['other 2003-06-11 home.xml', 'other 2003-08-15 work.xml'].forEach(n => {
      expect(passport.filter(n)).toBe(true);
   });

   [
      'Family-Name.mbox',
      'message.json',
      'Other - Voicemail - 2010-10-30T01_.mp3',
      'other 2004-06-05 work.xml',
      'other 2003-09-13 home.xml',
      'other3008@hotmail.com.xml'
   ].forEach(n => {
      expect(passport.filter(n)).toBe(false);
   });
});

test('parses file', async () => {
   const text = await readFileText(`${__dirname}/__mocks__/passport.xml`);
   const messages = passport.process(text, 'mock');

   expect(messages).toBeInstanceOf(Array);
   expect(messages).toHaveLength(4);
   expect(messages[0].from).toBe(Relation.Other);
   expect(messages[1].from).toBe(Relation.Self);
   expect(messages[3].text).toBe('sorry, been on and still on the phone');

   expect(messages).toMatchSnapshot();
});
