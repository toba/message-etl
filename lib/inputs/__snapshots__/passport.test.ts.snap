// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses file 1`] = `
Array [
  Object {
    "from": 1,
    "on": 2003-06-02T20:18:31.000Z,
    "source": 4,
    "text": "i have your mom in another window helping her with the printer",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:18:46.000Z,
    "source": 4,
    "text": "and two friends in two other windows",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:19:25.000Z,
    "source": 4,
    "text": "you missed the family chat yesterday, came on and four people started talking at once",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:19:30.000Z,
    "source": 4,
    "text": "i just need to make sure i type the write things in the right windows",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:19:42.000Z,
    "source": 4,
    "text": "fun--yeah, don't think i logged in yesterday",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:19:53.000Z,
    "source": 4,
    "text": "the computer was just rendering video so i did other stuff",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:20:13.000Z,
    "source": 4,
    "text": "how's that going?",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:20:44.000Z,
    "source": 4,
    "text": "slow but sure--good, your mom figured it out",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:22:05.000Z,
    "source": 4,
    "text": "tried to teach her a few things on how to add people to conversation etc. she just started chatting, Kayla is always on, we had to change names so she knew who was talking",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:22:22.000Z,
    "source": 4,
    "text": ":)",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:22:28.000Z,
    "source": 4,
    "text": "your mom?",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:23:15.000Z,
    "source": 4,
    "text": "no, kayla for the names, mom on the chatting lesson",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:23:42.000Z,
    "source": 4,
    "text": "right--saw amanda come on today but same story--very one sided conversation",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:23:52.000Z,
    "source": 4,
    "text": "okay, your mom is off to work again",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:24:49.000Z,
    "source": 4,
    "text": "she was for awhile yesterday, but I think she gets flustered trying to chat and look up things on the web at the same time",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:25:19.000Z,
    "source": 4,
    "text": "aren't you supposed to be working?",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:29:33.000Z,
    "source": 4,
    "text": "yes, am, just a sec",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:42:10.000Z,
    "source": 4,
    "text": "okay, pm was here talking about crazy marketing people",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:44:03.000Z,
    "source": 4,
    "text": "marketing people are almost as crazy as the people in I.T.",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:44:25.000Z,
    "source": 4,
    "text": "mom just called me a brat :'(",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:46:52.000Z,
    "source": 4,
    "text": "uh, i'm not gonna touch that one",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:47:08.000Z,
    "source": 4,
    "text": "sheesh, people keep talking to me, it's like i'm at work or something",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:47:17.000Z,
    "source": 4,
    "text": "chicken",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:48:24.000Z,
    "source": 4,
    "text": "i've been off yesterday and today, in the office both days, doing paperwork and faxing it over all day, it's cutting in on my pool time",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:48:48.000Z,
    "source": 4,
    "text": "i have much pity",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:49:35.000Z,
    "source": 4,
    "text": "you are so full of it",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:54:49.000Z,
    "source": 4,
    "text": ":)",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:54:59.000Z,
    "source": 4,
    "text": "i think i'm gonna go home early today--see ya",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:55:49.000Z,
    "source": 4,
    "text": "ok, now your a brat, going home early when you didn't work last week?!",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:56:27.000Z,
    "source": 4,
    "text": "i tire easily",
  },
  Object {
    "from": 2,
    "on": 2003-06-02T20:57:07.000Z,
    "source": 4,
    "text": "all the comments I could make to that, to easy though, talk to ya later",
  },
  Object {
    "from": 1,
    "on": 2003-06-02T20:57:26.000Z,
    "source": 4,
    "text": ":) bye",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T20:44:34.000Z,
    "source": 4,
    "text": "busy work day? :D",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T20:48:12.000Z,
    "source": 4,
    "text": "somewhat--shouldn't you be at work?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T20:52:12.000Z,
    "source": 4,
    "text": "should be, have been, but when a police officer wakes you up at 5am to tell you somebody through a rock through your window, it kinda messes up your day",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T20:54:42.000Z,
    "source": 4,
    "text": "should have specified, my car window",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T20:55:09.000Z,
    "source": 4,
    "text": "jeez, that's not good",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T20:55:18.000Z,
    "source": 4,
    "text": "scott gets back today, right?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T20:56:18.000Z,
    "source": 4,
    "text": "nope, he's stuck in Flagstaff in meetings until tommorrow, going to have a very grouchy wife to come home to",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T20:57:10.000Z,
    "source": 4,
    "text": "poor guy--did the police find the perpetrator?  was it random or did you piss someone off?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T20:58:59.000Z,
    "source": 4,
    "text": "they have it on video so who knows, there were three of us that got hit, they think one of the others pissed somebody off because of the video and mine was random to make it look that way, didn't even search my car",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T20:59:58.000Z,
    "source": 4,
    "text": "the CD player was stolen out of Kelly's car a few years back--they smashed up the dash pretty good getting it out--it's a lame thing to have happen",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:00:19.000Z,
    "source": 4,
    "text": "the police didn't bother to come by, though, even though i called them about it",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:01:35.000Z,
    "source": 4,
    "text": "there's that ajflores again, let's see if i get a response this time ...",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:01:42.000Z,
    "source": 4,
    "text": "yeah, doubt it's a priority for them especially for Phoenix, between that and my tire splitting I think I need to leave again",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:01:51.000Z,
    "source": 4,
    "text": "good luck I've tried twice",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:01:59.000Z,
    "source": 4,
    "text": "this is my third attempt :)",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:02:11.000Z,
    "source": 4,
    "text": "maybe i need to say something more provocative",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:02:22.000Z,
    "source": 4,
    "text": "such as?",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:02:29.000Z,
    "source": 4,
    "text": "that's the problem nothing comes to mind",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:05:16.000Z,
    "source": 4,
    "text": "'./mmm,mm gjl.",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:05:23.000Z,
    "source": 4,
    "text": "huh?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:06:22.000Z,
    "source": 4,
    "text": "sorry, had a phone call, Hunter wanted to talk I guess, turning off the phone, work driving me crazy",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:11:36.000Z,
    "source": 4,
    "text": "no prob, Sunil from India was talking to me anyway",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:11:49.000Z,
    "source": 4,
    "text": "we have a crew of Indians here lately",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:17:18.000Z,
    "source": 4,
    "text": "bathroom break ...",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:29:18.000Z,
    "source": 4,
    "text": "damn, wasn't quick enough with the phone, plus forgot my second latte in the car, still hot though",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:31:40.000Z,
    "source": 4,
    "text": "you drink hot lattes in Phoenix!?",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:32:03.000Z,
    "source": 4,
    "text": "do you like a flavor (for reference next time i buy you one on an early monday in forks)?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:33:40.000Z,
    "source": 4,
    "text": "Irish Cream, it's an addiction can't live without it French Vanilla a second choice. I wouldn't function without one a day",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:35:02.000Z,
    "source": 4,
    "text": "timamisu for me, though not daily.  sheesh, i'm not a druggy like that",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:35:14.000Z,
    "source": 4,
    "text": "gotta go chat on the couch a sec about some tech stuff ..",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:35:39.000Z,
    "source": 4,
    "text": "goodluck",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:54:28.000Z,
    "source": 4,
    "text": "that was fun, though they're always disgusted by my swigs of spicy v8 juice",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:55:28.000Z,
    "source": 4,
    "text": "would have to agree, love the v8 not the spicy though",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:55:40.000Z,
    "source": 4,
    "text": "wow, how boring to agree",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:56:34.000Z,
    "source": 4,
    "text": "surprised you though, enjoy the rare occasions it happens :D",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:57:29.000Z,
    "source": 4,
    "text": "well, i sang a lionel richie song to amanda through im and still no response, so i'm guessing something must be broken",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:58:03.000Z,
    "source": 4,
    "text": "jeez, just noticed that timamisu above should be tiramisu",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:58:21.000Z,
    "source": 4,
    "text": "i also refer to it as the dusty barn flavor",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:58:36.000Z,
    "source": 4,
    "text": "we were thinking of opening a brewery here at work with that theme",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:58:50.000Z,
    "source": 4,
    "text": "beers like dusty barn, rusty nail, squeeking hinge and such",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:59:15.000Z,
    "source": 4,
    "text": "knew what you meant, these are the tech converstions you mentioned?",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T21:59:38.000Z,
    "source": 4,
    "text": "uh, no, we talked about some programming stuff and then x-men (who has the best power)",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T21:59:55.000Z,
    "source": 4,
    "text": "and your thought on that?",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:00:25.000Z,
    "source": 4,
    "text": "magneto's power is cool, but we couldn't decide if it was better than captain picard's (we still call him that from star trek)",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:00:48.000Z,
    "source": 4,
    "text": "we thought poison girl's power was a bit lame",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:01:22.000Z,
    "source": 4,
    "text": "wolverine obviously has sex appeal but the powers that operate over distances seem better",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:01:41.000Z,
    "source": 4,
    "text": "ok, ive had ALOT of coffee and you're making me laugh, not a good combo",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:02:14.000Z,
    "source": 4,
    "text": "sometimes we get laughing too much here too",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:02:23.000Z,
    "source": 4,
    "text": "some of the managers in this building have complained about it",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:03:23.000Z,
    "source": 4,
    "text": "we also talked about brett, and how crazy he is",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:03:31.000Z,
    "source": 4,
    "text": "it's a jealousy thing because they aren't getting to",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:03:49.000Z,
    "source": 4,
    "text": "could be--some do seem to be sourpusses",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:03:59.000Z,
    "source": 4,
    "text": "anything knew on him or the same stuff",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:04:25.000Z,
    "source": 4,
    "text": "we went to my house for lunch one day and brett was there (staying with us a week) with his shirt off, so now he's forever remembered as the shirtless guy at my house",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:04:41.000Z,
    "source": 4,
    "text": "he's in maryland now at a clinic doing a 21 day fast",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:04:59.000Z,
    "source": 4,
    "text": "purging those bad toxins, physical and mental, i guess",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:05:25.000Z,
    "source": 4,
    "text": "he doesn't have enough energy to sit up much so he just lays around, sometimes out in the sun",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:06:03.000Z,
    "source": 4,
    "text": "he's planning to live out of his camper van as soon as he gets a trailer for pulling his car around",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:06:46.000Z,
    "source": 4,
    "text": "i think he was also trying to use hypnotic regression to find the reasons in past lives for some of his present phobias",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:07:02.000Z,
    "source": 4,
    "text": "okay, did you get any of that?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:07:11.000Z,
    "source": 4,
    "text": "whatever happened, I did'nt do it",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:07:37.000Z,
    "source": 4,
    "text": "s .. u .. r .. e     caffeine girl",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:08:06.000Z,
    "source": 4,
    "text": "ok, your still in comic book mode",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:08:19.000Z,
    "source": 4,
    "text": "i talked about brett for five pages but i guess you were off then",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:08:26.000Z,
    "source": 4,
    "text": "as bad as amanda",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:08:52.000Z,
    "source": 4,
    "text": "gonna go so matrix tonight, finally",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:08:56.000Z,
    "source": 4,
    "text": "that was so uncalled for, by the way you're a bad liar",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:05.000Z,
    "source": 4,
    "text": "crap--&quot;go see&quot;",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:26.000Z,
    "source": 4,
    "text": "he's in maryland now at a clinic doing a 21 day fast",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:26.000Z,
    "source": 4,
    "text": "purging those bad toxins, physical and mental, i guess",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:26.000Z,
    "source": 4,
    "text": "he doesn't have enough energy to sit up much so he just lays ar...",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:26.000Z,
    "source": 4,
    "text": "he's planning to live out of his camper van as soon as he gets ...",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:26.000Z,
    "source": 4,
    "text": "i think he was also trying to use hypnotic regression to find t...",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:26.000Z,
    "source": 4,
    "text": "okay, did you get any of that?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:09:31.000Z,
    "source": 4,
    "text": "how much have you had?(coffee)",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:49.000Z,
    "source": 4,
    "text": "maybe it was five sentences, not five pages",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:57.000Z,
    "source": 4,
    "text": "no coffee for me, too warm out",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:09:59.000Z,
    "source": 4,
    "text": "just the spicy v8",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:10:03.000Z,
    "source": 4,
    "text": "like drinking salsa",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:10:05.000Z,
    "source": 4,
    "text": "i like salsa",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:10:51.000Z,
    "source": 4,
    "text": "maybe your'e frying something, spelling is getting bad!",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:11:40.000Z,
    "source": 4,
    "text": "i just need some beer to clear my brain plagues",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:11:45.000Z,
    "source": 4,
    "text": "(at least that's what we always say here)",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:11:58.000Z,
    "source": 4,
    "text": "yuck!!!",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:12:03.000Z,
    "source": 4,
    "text": "brain plagues are thought to be the cause of alzheimers",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:12:32.000Z,
    "source": 4,
    "text": "has work slowed down--wasn't a dm coming this week?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:13:51.000Z,
    "source": 4,
    "text": "no picked up, she was here Friday, rm coming next tue corporate lp 16th but nobody but me knows that, so freaking out cause I'm not there",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:14:32.000Z,
    "source": 4,
    "text": "i'm sure the caffeine will calm you down :)",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:14:40.000Z,
    "source": 4,
    "text": "are you taking the rest of today off, or working from home?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:15:04.000Z,
    "source": 4,
    "text": "it's the only thing keeping me going, even my dm knows that",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:16:24.000Z,
    "source": 4,
    "text": "both? been in the office, had to reschedule meetings and touch base with dm, leave a to do list for staff,been doing work from home",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:17:56.000Z,
    "source": 4,
    "text": "sounds hectic--are you hoping you're as yet undiscovered career goals will afford a slower pace of life?",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:18:08.000Z,
    "source": 4,
    "text": "(you're should be your)",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:19:21.000Z,
    "source": 4,
    "text": "unsure, that's kind of what I'm used to so/",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:20:17.000Z,
    "source": 4,
    "text": "does &quot;used to&quot; = &quot;prefer&quot;?",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:20:23.000Z,
    "source": 4,
    "text": "hunter is ready for the pool so, enjoy the movie, let me know if it's good",
  },
  Object {
    "from": 1,
    "on": 2003-06-03T22:20:30.000Z,
    "source": 4,
    "text": "have fun!",
  },
  Object {
    "from": 2,
    "on": 2003-06-03T22:20:34.000Z,
    "source": 4,
    "text": "pretty much",
  },
  Object {
    "from": 2,
    "on": 2003-06-05T15:57:50.000Z,
    "source": 4,
    "text": "Good morning! Have a great day, on my way out the door. Don't overwork yourself today :D",
  },
  Object {
    "from": 1,
    "on": 2003-06-05T15:58:01.000Z,
    "source": 4,
    "text": "thanks, have fun",
  },
]
`;
